pkgs/dart_model/lib/src/json_buffer.dart

```diff
@@ -0,0 +1,390 @@
+import 'dart:collection';
+import 'dart:convert';
+import 'dart:typed_data';
+
+/// Map with lazily evaluated values.
+///
+/// This is the efficient way to accumulate data into a [JsonBuffer]. Nested
+/// maps should also be `LazyMap`.
+class LazyMap with MapMixin<String, Object?> implements Map<String, Object?> {
+  @override
+  final List<String> keys;
```

Comment:
Can we keep this as an `Iterable<String>`? Those are already unmodifiable.

Reply:
Done, added a private `List` field to avoid a cast later on in `addMap`.